hpx/runtime/serialization/detail/vc.hpp

```diff
@@ -86,6 +90,39 @@ namespace hpx { namespace serialization
     }
 }}
 
+#else
+
+#include <array>
+
+#include <Vc/datapar>
+
+///////////////////////////////////////////////////////////////////////////////
+namespace hpx { namespace serialization
+{
+    template <typename T, typename Abi>
+    void serialize(input_archive & ar, Vc::datapar<T, Abi>& v, unsigned)
+    {
+        std::array<T, Vc::datapar<T, Abi>::size()> data;
+        std::size_t size = 0;
+        ar & size & data;
+        HPX_ASSERT(size == Vc::datapar<T, Abi>::size());
+
+        v.copy_from(data.data(), Vc::vector_aligned);
+    }
+
+    template <typename T, typename Abi>
+    void serialize(output_archive & ar, Vc::datapar<T, Abi> const& v, unsigned)
+    {
+        std::array<T, Vc::datapar<T, Abi>::size()> data;
+        std::size_t size = Vc::datapar<T, Abi>::size();
+        v.copy_to(data.data(), Vc::vector_aligned);
+
+        ar & size & data;
```

Comment:
Do we really need to send along the size here? Isn't it a compile constant in any case?